main.js

```diff
@@ -119,48 +147,78 @@ function main() {
     Logger.log(matcher.allStudents[0].firstName);
     Logger.log(matcher.allStudents[0].preferences[0].name);
 
-    populateSheet(outputSheet, matcher);
+    populateSheet(outputSheet, dataSheet, matcher);
 }
 
+
 /**
  * Output the results of the matcher to the given sheet.
  */
-function populateSheet(outputSheet, matcher) {
+function populateSheet(outputSheet, dataSheet, matcher) {
+    //Formats the sheets before writing to them.
     outputSheet.clear();
     outputSheet.appendRow(OUTPUT_SHEET_HEADERS);
     outputSheet.setFrozenRows(1);
 
-    matcher.fixStudentPreferences();
+  
+    dataSheet.clear();
+    dataSheet.appendRow(DATA_SHEET_HEADER1);
+    outputSheet.setFrozenRows(1);
 
+    matcher.fixStudentPreferences();
     matcher.matchGirls();
 
-    // All student lines to output
+
+    var results = [0, 0, 0, 0, 0, 0, 0];
     var studentLines = [];
 
     for (var i = 0; i < matcher.allStudents.length; i++) {
         var student = matcher.allStudents[i];
+        var prefnums = student.checkPreferenceNumbers();
+        Logger.log(results);
+        Logger.log(prefnums);
+        //For loop records the the number of times a student has # preference in his preference list
+        for (var z = 0; z < prefnums.length; z++) {
+            results[prefnums[z]] = results[prefnums[z]] + 1;
+        }
+
         var studentLine = [];
         studentLine.push(student.firstName);
         studentLine.push(student.lastName);
         studentLine.push(student.grade);
 
         // List the student's assigned workshops in the row
         for (var j = 0; j < student.assignedWorkshops.length; j++) {
-            var workshop = student.assignedWorkshops[j];
-            if (workshop === null) {
-                studentLine.push("", "", "");
-            } else {
-                studentLine.push(workshop.number);
-                studentLine.push(workshop.name);
-                studentLine.push(workshop.location);
-            }
+            var assignedWorkshop = student.assignedWorkshops[j];
+            studentLine.push(assignedWorkshop.number);
+            studentLine.push(assignedWorkshop.name);
+            studentLine.push(assignedWorkshop.location);
+            //studentLine.push(workshop.toString());
+            //Logger.log(studentLine);
         }
         studentLines.push(studentLine);
     }
 
+
+    for (var k = 0; k < matcher.workshopsByPopularity.length; k++) {
+        Logger.log(matcher.workshopsByPopularity.length);
+        var dataInfo = [];
```

Comment:
Can `dataInfo` be called something more meaningful? The variable name isn't clear.